macros/directory_pattern.sql

```diff
@@ -0,0 +1,24 @@
+-- these macros will read a user’s home environment and detect whether a computer’s operating system is Windows based or Mac/Linux, and display the right directory pattern.
+{% macro directory_pattern() %}
+  {%- if env_var("HOME", "not_set") != "not_set" -%}
```

Comment:
What does the existence of the `HOME` env var tell us? It would be useful to have a comment here explaining what this proves Is it that it's a Mac/Linux machine as opposed to Windows?

If so, then maybe instead you give it semantic meaning: 

```
{% set is_windows = env_var("HOME". "not_set") != "not_set" %}
{% if is_windows %}
  [do the windows stuff]
{% else %}
  [do the everything else stuff]
{% endif %}
```

Reply:
The thing is that in the vast majority of cases, `$HOME` will be set only for Mac/Linux, but this code is also checking that if it is defined and contains a `\` we are on Windows. It could be the case that people on Windows have defined their own `$HOME` env var.

So, while `is_windows` might not be the name to use I agree that we could provide more context throughthe variable names.

Reply:
Not sure which variables should be renamed based on this conversation, so I would love more clarification. I do have a quick question. Would it be better if this macro was written with an `and` condition combining the nested `if statements`?:

`  {% macro get_directory_pattern() %}`
---` {%- if env_var("HOME", "not_set") != "not_set" and "\\\\" not in env_var("HOME", "not_set") -%}`
------`{{ return("/") }}`
---`{%- else -%}`
------`{{ return("\\\\") }}`
---`{%- endif -%}`
`{% endmacro %}`

^ this would be better for less code 🤔

Reply:
What you could do here Angelica maybe would be to:
- create a variable called `env_var_home_exists` which has the value `env_var("HOME", "not_set") != "not_set"` ; which would tell us if the env var exists or not
- then, we can check `env_var("HOME")` without the second arg in the nested if (because we know that the env var is set) when we want to check if it contains `\`

```
{% macro directory_pattern() %}
  {%- set  env_var_home_exists = env_var("HOME", "not_set") != "not_set" -%}
  {%- if env_var_home_exists -%}
     {% if "\\\\" not in env_var("HOME") %}
       {{ return("/") }}
     {% else %}
       {{ return("\\\\") }}
     {% endif %}
  {%- else -%}
      {{ return("\\\\") }}
  {%- endif -%}
{% endmacro %}
```

Reply:
@b-per this works perfectly!